### PR TITLE
feat: added add/update operations for note

### PR DESCRIPTION
**Pull request by KanekarSnehal**
In this PR completed add and update operation.
user can add new note using create note button and edit/update the existing note using edit icon button.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -54,3 +55,11 @@
   width: 2.8rem;
   cursor: pointer;
 }
+
+/* .active-icon{
```
Comment (ArjunGTX) [inline src/App.css:59]: Remove commented code

File: src/components/Dropdown/Dropdown.jsx [hunk 2]
```diff
@@ -1,8 +1,18 @@
 import "./label.css";
+import { useNotes } from "../../context/index";
```
Comment (ArjunGTX) [inline src/components/Dropdown/Dropdown.jsx:2]: `"../../context"` is enough

File: src/components/ModalInput/ModalInput.jsx [hunk 3]
```diff
@@ -1,12 +1,65 @@
 import { Palette, Dropdown } from "../index";
-import { useModal } from "../../context/index";
+import { useModal, useNotes } from "../../context/index";
```
Comment (ArjunGTX) [inline src/components/ModalInput/ModalInput.jsx:2]: `"../../context"` is enough

File: src/components/ModalInput/ModalInput.jsx [hunk 4]
```diff
@@ -1,12 +1,65 @@
...
+  const handleAddNote = async () => {
+    try {
+      const { data } = await addNote(noteInfo);
+      noteDispatch({ type: "NOTE_ADDED", payload: data.notes });
+      console.log(data.notes);
+    } catch (error) {}
+  };
+
+  const handleUpdateNote = async () => {
+    // console.log(noteData, noteData.id);
```
Comment (ArjunGTX) [inline src/components/ModalInput/ModalInput.jsx:54]: Remove all commented code

File: src/components/NoteCards/NoteCards.jsx [hunk 5]
```diff
@@ -1,44 +1,129 @@
...
     <>
       <div className="notes-keeping-area ">
-        {data &&
-          data.map((dataItem) => (
-            <div className="note-input-container grid-item-note">
+        {notes.length !== 0 &&
+          notes.map((dataItem) => (
+            <div
+              className="note-input-container grid-item-note "
+              style={{ backgroundColor: dataItem.noteColor }}
```
Comment (ArjunGTX) [inline src/components/NoteCards/NoteCards.jsx:33]: Refrain from using inline styles. Use css classes instead

File: src/components/Palette/Palette.jsx [hunk 6]
```diff
@@ -15,24 +17,42 @@ const palette = [
   "#E8EAED",
 ];
 
-export const Palette = () => {
+export const Palette = ({ dataItem, type, setNoteInfo }) => {
   const [isHidden, setIsHidden] = useState(true);
+  const { notes, noteData, noteDispatch } = useNotes();
+  // console.log(noteData);
```
Comment (ArjunGTX) [inline src/components/Palette/Palette.jsx:23]: There are `console.log()` s commented in almost all the files. Remove them after debugging

**Review by ArjunGTX — CHANGES_REQUESTED** [review]

**Review by spraveenofficial — APPROVED** [review]

File: src/components/NoteCards/NoteCards.jsx [hunk 7]
```diff
@@ -1,44 +1,129 @@
...
-import { useModal } from "../../context/index";
+import { useModal, useNotes } from "../../context/index";
 import { ModalInput } from "../ModalInput/ModalInput";
+import { useEffect } from "react";
+import { addNote, updateNote } from "../../services/noteServices";
 
 export const NoteCards = ({ data }) => {
   const { modalState, modalDispatch } = useModal();
+  const { notes, noteData, noteDispatch, currNote } = useNotes();
+  // console.log(notes);
```
Comment (deepak29-git) [inline src/components/NoteCards/NoteCards.jsx:10]: remove commented code

File: src/pages/userAuth/Login.jsx [hunk 8]
```diff
@@ -25,7 +25,7 @@ export const Login = () => {
         password: userInfo.password,
       });
 
-      console.log(data.user);
+      // console.log(data.user);
```
Comment (deepak29-git) [inline src/pages/userAuth/Login.jsx:28]: remove console after debugging

File: src/services/noteServices.jsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+const config = {
+  headers: { authorization: localStorage.getItem("token") },
+};
+
+export const getNotes = async () => {
+  return await axios.get(`/api/notes`, config);
+};
+
+export const addNote = async (note) => {
```
Comment (deepak29-git) [inline src/services/noteServices.jsx:11]: this is good , I like this approach

File: src/services/noteServices.jsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  return await axios.post(`/api/notes`, { note }, config);
+};
+
+export const updateNote = async (noteID, note) => {
+  // console.log("servvvvvvvvvvvvvvvvv");
```
Comment (deepak29-git) [inline src/services/noteServices.jsx:16]: remove console

**Review by deepak29-git — APPROVED** [review]

File: src/services/noteServices.jsx (hunk 10 above)
Comment (KanekarSnehal) [reply src/services/noteServices.jsx]: Done

File: src/pages/userAuth/Login.jsx (hunk 8 above)
Comment (KanekarSnehal) [reply src/pages/userAuth/Login.jsx]: Done

File: src/components/NoteCards/NoteCards.jsx (hunk 5 above)
Comment (KanekarSnehal) [reply src/components/NoteCards/NoteCards.jsx:29]: It's used to change the container color, when user choose color from palette.

File: src/components/ModalInput/ModalInput.jsx (hunk 4 above)
Comment (KanekarSnehal) [reply src/components/ModalInput/ModalInput.jsx]: Done

File: src/App.css (hunk 1 above)
Comment (KanekarSnehal) [reply src/App.css]: Done

**Pull request merged**